cli/azd/cmd/middleware/command_hooks.go

```diff
@@ -0,0 +1,86 @@
+package middleware
```

Comment:
should we call it azdmidleware?  
Also, should it be within /cmd folder?  I have the impression that /cmd is the cobra layer in the very top.  I wonder if the midleware should be on /pkg

related:  I just noticed that the package `actions` is on /cmd.  That also looks weird, it gives the impression that the Action is something which only makes sense for the cobra layer. I was under the impression that an Action could be used without cobra.

Reply:
Right now our `Action` interface is in `cmd/actions` and didn't want to change too much for this POC.  The middleware chain is invoked on the actions so made sense to keep it in the same package for the time being.